modules/vba_documents/app/models/vba_documents/upload_submission.rb

```diff
@@ -25,5 +46,24 @@ def signed_url(guid)
       obj = s3.bucket(Settings.vba_documents.s3.bucket).object(guid)
       obj.presigned_url(:put, {})
     end
+
+    def status_in_flight?
+      IN_FLIGHT_STATUSES.include?(status)
+    end
+
+    def map_downstream_status(body)
+      response_object = JSON.parse(body)[0][0]
+      if response_object['status'] == 'Received'
+        self.status = 'received'
+      elsif response_object['status'] == 'In Process'
+        self.status = 'processing'
+      elsif response_object['status'] == 'Success'
+        self.status = 'success'
+      elsif response_object['status'] == 'Error'
+        self.status = 'error'
+        self.code = 'DOC202'
+        self.detail = "Downstream status: #{response_object['errorMessage']}"
+      end
```

Comment:
All documented statuses accounted for; potentially worth tossing an exception if we're not able to map the status.

Reply:
Done